x-pack/plugins/actions/server/lib/gen_ai_token_tracking.ts

```diff
@@ -98,13 +98,14 @@ export const getGenAiTokenTracking = async ({
     (validatedParams.subAction === 'run' || validatedParams.subAction === 'test')
   ) {
     try {
+      const rData = result.data as unknown as {
```

Comment:
You might want to consider doing more validation on this - as well as the structure below.  If for some reason, the data isn't shaped the way you expect, it may be difficult to diagnose.  I think I'd write a little function to parse the data and return as a typed structure or undefined (on error), then log the shape on error.

Reply:
We have pretty big diverse of the the possible return types here. ++ to consider some validation approach later, because as more GenAI integrations we will have the more options/handlers we get. Maybe borrow some ideas from LangChain or other similar solutions.

Reply:
i added it!

Reply:
https://github.com/elastic/kibana/pull/179304/commits/0c0c6fc2489c726117358a6b8b4cae7d10ffa693